File: ios/HelloMessage.m
//
//  HelloMessage.m
//  SampleProject
//
//  Created by Samik Roy on 16/06/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(HelloMessage, NSObject)

RCT_EXTERN_METHOD(sayHello:(NSString *)name
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/HelloMessage.swift
//
//  HelloMessage.swift
//  SampleProject
//
//  Created by Samik Roy on 16/06/25.
//

import Foundation
import React

@objc(HelloMessage)
class HelloMessage: NSObject {
  @objc
  func sayHello(_ name: String, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    resolve("Hello \(name) from Swift!")
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
